src/vector/_pytree.py

```diff
@@ -0,0 +1,96 @@
+"""PyTree operations for vector objects.
+
+This module defines how vector objects are handled within optree.
+See https://blog.scientific-python.org/pytrees/ for the rationale for these functions.
+"""
+
+from __future__ import annotations
+
+from types import ModuleType
+from typing import Any
+
+from vector._methods import (
+    Vector2D,
+    Vector3D,
+    Vector4D,
+)
+from vector.backends.object import (
+    MomentumObject2D,
+    MomentumObject3D,
+    MomentumObject4D,
+    VectorObject2D,
+    VectorObject3D,
+    VectorObject4D,
+)
```

Comment:
I *think* we can also add numpy backends, as long as we also register the numpy array as a pytree node.